src/bar.cpp

```diff
@@ -579,6 +587,13 @@ waybar::Bar::Bar(struct waybar_output* w_output, const Json::Value& w_config)
   // GTK layer shell anchors logic relying on the dimensions of the bar.
   surface_impl_->setPosition(position);
 
+  hotspotsurface_impl_->setMargins(margins_);
```

Comment:
Note that the bar doesn't have to be adjacent to the edge of the screen. Non-zero margin on an anchored edge may break your hotspot logic.

Reply:
@alebastr , when waybar is docked right or left and when it doesn't occupy whole screen height, I thought that margins helped in that case. Will correct it.

Reply:
@alebastr , have removed it